prov/gni/src/gnix_av.c

```diff
@@ -371,15 +380,37 @@ static int map_reverse_lookup(struct gnix_fid_av *av_priv,
 {
 	GNIX_HASHTABLE_ITERATOR(av_priv->map_ht, iter);
 	struct gnix_av_addr_entry *entry;
+	fi_addr_t rx_addr;
 
 	while ((entry = _gnix_ht_iterator_next(&iter))) {
-		if (GNIX_ADDR_EQUAL(entry->gnix_addr, gnix_addr)) {
-			*fi_addr = GNIX_HASHTABLE_ITERATOR_KEY(iter);
-			return FI_SUCCESS;
+		/*
+		 * for SEP endpoint entry we may have a delta in the cdm_id
+		 * component of the address to process
+		 */
+		if ((entry->name_type & GNIX_EPN_TYPE_SEP) &&
+		    (entry->gnix_addr.device_addr == gnix_addr.device_addr)) {
+			if (gnix_addr.cdm_id - entry->gnix_addr.cdm_id <
+					entry->rx_ctx_cnt) {
+				/* we have a match */
+				int ctx_bits = 0;
+
+				rx_addr = *(fi_addr_t *)&entry->gnix_addr;
+
+				while (entry->rx_ctx_cnt >> ++ctx_bits);
```

Comment:
We talked about this.  It seems like we should be using av->rx_ctx_bits to generate the fi_rx_addr.

That also means we're missing some other necessary changes.  fi_addr_t's belong to the local AV  and should use a format matching the rx_ctx_bits in the AV they are being inserted into.  That means that when inserting addresses into an AV, we need to check that the rx_ctxs in a gnix_ep_name being inserted can fit in the AV's rx_ctx_bits.  Also, we need to create the output fi_addr_t using the rx_ctx_bits specified in the AV we are inserting into.